GitHub_LearningTests/Feature/Onboarding/Presentation/OnboardingPresenterTests.swift

```diff
@@ -28,78 +32,72 @@ class OnboardingPresenterTests: XCTestCase {
     
     func testIncresePage_whenNextNotOnLastPage() {
         // Given
-        let view = FakeView()
+        let controller = FakeOnboardingController()
+        let view = FakeOnboardingView()
         let launcher = FakeLoginLauncherController()
-        setUpWith(onboardingLauncher: launcher, onboardingView: view)
+        setUpWith(onboardingController: controller, onboardingLauncher: launcher, onboardingView: view)
         
         // When
-        sut.viewDidLoad()
-        var nextPage = 0
-        view.presentingIndex = 0
-        for _ in 0..<view.pages.count - 1 {
-            nextPage = view.presentingIndex! + 1
-            sut.next()
-        }
-        
+        let index = 0
+        sut.index = index
+        sut.next()
+            
         // Then
-        XCTAssertTrue(view.presentingIndex == nextPage)
+        XCTAssertTrue(view.isPageShown)
+        XCTAssertEqual(view.presentingIndex, index + 1)
         XCTAssertNil(launcher.isLaunched)
     }
     
     func testLaunchLogin_whenNextOnLastPage() {
         // Given
+        let controller = FakeOnboardingController()
         let launcher = FakeLoginLauncherController()
-        let view = FakeView()
-        setUpWith(onboardingLauncher: launcher, onboardingView: view)
+        let view = FakeOnboardingView()
+        setUpWith(onboardingController: controller, onboardingLauncher: launcher, onboardingView: view)
         
         // When
-        sut.viewDidLoad()
-        for _ in 0...view.pages.count - 1 {
-            sut.next()
-        }
+        let index = controller.lastPageIndex
+        sut.index = index
+        sut.next()
         
         // Then
         XCTAssertNotNil(launcher.isLaunched)
     }
     
     func testViewUpdate_whenSelectedPage() {
         // Given
-        let view = FakeView()
+        let view = FakeOnboardingView()
         setUpWith(onboardingView: view)
-        
+        let index = 2
+
         // When
-        sut.viewDidLoad()
-        let index = Int.random(in: 0...view.pages.count - 1)
         sut.selectedPage(at: index)
         
-        
         // Then
-        XCTAssertNotNil(view.presentingIndex)
+        XCTAssertEqual(view.presentingIndex, index)
     }
     
     func testButtonTitleUpdate_whenSelectedPageIsLastPage() {
         // Given
-        let view = FakeView()
-        setUpWith(onboardingView: view)
+        let controller = FakeOnboardingController()
+        let view = FakeOnboardingView()
+        setUpWith(onboardingController: controller, onboardingView: view)
         
         // When
-        sut.viewDidLoad()
-        sut.selectedPage(at: view.pages.count - 1)
-        
+        sut.selectedPage(at: controller.lastPageIndex)
```

Comment:
Nady primo nastav ten index. Testujes `index` -> `button title` a ne `selectedPage` -> `button title`, i kdyz je to dusledek te funkce.

Reply:
no já jsem myslel že tady jsem chtěli otestovat, že funkce selectedPage funguje a dělá co má...K tomu indexu jsme se přece chtěli dostat, abych nemusel 4x volat next.